src/proj.rs

```diff
@@ -592,6 +593,28 @@ impl Proj {
         transform_epsg(ctx, from, to, area)
     }
 
+    /// Return Authority code for Proj
+    ///
+    /// # Safety
+    /// This method contains unsafe code.
+    pub fn id_code(&self) -> Result<&str, str::Utf8Error> {
+        let pj = self.c_proj as *const PJ;
+        let c_char = unsafe { proj_get_id_code(pj, 0) };
+        let c_str = unsafe { CStr::from_ptr(c_char) };
```

Comment:
I believe there should be nullptr checking here as proj_get_id_code() might return null in case of error (typically if the CRS is custom and not tied to a known code), and https://doc.rust-lang.org/std/ffi/struct.CStr.html#method.from_ptr indicates the CStr::from_ptr() shouldn't be called with a null ptr

Reply:
When testing, I found that the Proj created by proj4str and esriwkt could not recognize the code and name。

Reply:
> I found that the Proj created by proj4str and esriwkt could not recognize the code and name。

Yes, that's fully expected. Not all CRS representations contain an authority name and code

Reply:
If there are corresponding ideas or methods, converting them would be ideal.

Reply:
you can try using proj_identify() to find a known CRS that is identical to the current one, but there's no guarantee it will succeed. Custom CRS with custom projection parameter values are fully valid